ui/apps/akasha/src/components/pages/entry-page/beam-page.tsx

```diff
@@ -149,29 +151,34 @@ const BeamPage: React.FC<unknown> = () => {
                 reflectToId={beamId}
                 contentClickable={true}
                 onContentClick={() =>
-                  navigateTo({
-                    appName: '@akashaorg/app-akasha-integration',
-                    getNavigationUrl: navRoutes => `${navRoutes.Reflect}/${itemData.node.id}`,
+                  navigate({
+                    to: `/reflection/$reflectionId`,
+                    params: {
+                      reflectionId: itemData.node.id,
+                    },
                   })
                 }
                 onReflect={() =>
-                  navigateTo({
-                    appName: '@akashaorg/app-akasha-integration',
-                    getNavigationUrl: navRoutes =>
-                      `${navRoutes.Reflect}/${itemData.node.id}${navRoutes.Reflect}`,
+                  navigate({
+                    to: `/reflection/$reflectionId/reflect`,
```

Comment:
better import and use the routes already defined, something like ${routes[REFLECTION]}/$reflectionId${routes[REFLECT]}

Reply:
TypeScript provides the route and it's all safe, this is not hardcoded route :) ... if you try the approach you shared it won't work. It is the same for all.